client/client-multi/client-java-core/src/main/java/org/kaaproject/kaa/client/logging/DefaultLogUploadStrategy.java

```diff
@@ -26,32 +26,39 @@
 public class DefaultLogUploadStrategy implements LogUploadStrategy {
     private static final Logger LOG = LoggerFactory.getLogger(DefaultLogUploadStrategy.class);
 
-    private static final int DEFAULT_UPLOAD_TIMEOUT = 2 * 60;
-    private static final int DEFAULT_UPLOAD_CHECK_PERIOD = 30;
-    private static final int DEFAULT_RETRY_PERIOD = 5 * 60;
-    private static final int DEFAULT_UPLOAD_VOLUME_THRESHOLD = 8 * 1024;
-    private static final int DEFAULT_UPLOAD_COUNT_THRESHOLD = 64;
-    private static final int DEFAULT_BATCH_SIZE = 8 * 1024;
-    private static final int DEFAULT_BATCH_COUNT = 256;
-
-    private int timeout = DEFAULT_UPLOAD_TIMEOUT;
-    private int uploadCheckPeriod = DEFAULT_UPLOAD_CHECK_PERIOD;
-    private int retryPeriod = DEFAULT_RETRY_PERIOD;
-    private int volumeThreshold = DEFAULT_UPLOAD_VOLUME_THRESHOLD;
-    private int countThreshold = DEFAULT_UPLOAD_COUNT_THRESHOLD;
-    private int batchSize = DEFAULT_BATCH_SIZE;
-    private int batchCount = DEFAULT_BATCH_COUNT;
+    protected static final int DEFAULT_UPLOAD_TIMEOUT = 2 * 60;
+    protected static final int DEFAULT_UPLOAD_CHECK_PERIOD = 30;
+    protected static final int DEFAULT_RETRY_PERIOD = 5 * 60;
+    protected static final int DEFAULT_UPLOAD_VOLUME_THRESHOLD = 8 * 1024;
+    protected static final int DEFAULT_UPLOAD_COUNT_THRESHOLD = 64;
+    protected static final int DEFAULT_BATCH_SIZE = 8 * 1024;
+    protected static final int DEFAULT_BATCH_COUNT = 256;
+    protected static final int DEFAULT_TIME_LIMIT = 5 * 60;
+    protected static final boolean DEFAULT_UPLOAD_LOCKED = false;
+
+
+    protected int timeout = DEFAULT_UPLOAD_TIMEOUT;
+    protected int uploadCheckPeriod = DEFAULT_UPLOAD_CHECK_PERIOD;
+    protected int retryPeriod = DEFAULT_RETRY_PERIOD;
+    protected int volumeThreshold = DEFAULT_UPLOAD_VOLUME_THRESHOLD;
+    protected int countThreshold = DEFAULT_UPLOAD_COUNT_THRESHOLD;
+    protected int batchSize = DEFAULT_BATCH_SIZE;
+    protected int batchCount = DEFAULT_BATCH_COUNT;
+    protected boolean UPLOAD_LOCKED = DEFAULT_UPLOAD_LOCKED;
     
 
     @Override
     public LogUploadStrategyDecision isUploadNeeded(LogStorageStatus status) {
         LogUploadStrategyDecision decision = LogUploadStrategyDecision.NOOP;
-        if (status.getConsumedVolume() >= volumeThreshold) {
-            LOG.info("Need to upload logs - current size: {}, threshold: {}", status.getConsumedVolume(), volumeThreshold);
-            decision = LogUploadStrategyDecision.UPLOAD;
-        }else if (status.getRecordCount() >= countThreshold) {
-            LOG.info("Need to upload logs - current count: {}, threshold: {}", status.getRecordCount(), countThreshold);
-            decision = LogUploadStrategyDecision.UPLOAD;
+
+        if(!UPLOAD_LOCKED) {
```

Comment:
It is better to create protected method checkUploadNeeded with same signature. This will allow to overwrite checkUploadNeeded and avoid duplication of this if statement